src/lib/support/jsontlv/TlvToJson.cpp

```diff
@@ -0,0 +1,371 @@
+        VerifyOrReturnError(byteString.Get() != nullptr, CHIP_ERROR_NO_MEMORY);
+
+        auto encodedLen              = Base64Encode(span.data(), static_cast<uint16_t>(span.size()), byteString.Get());
+        byteString.Get()[encodedLen] = '\0';
+
+        InsertKeyValue(jsonObj, context, byteString.Get());
+        break;
+    }
+
+    case TLV::kTLVType_UTF8String: {
+        CharSpan span;
+        ReturnErrorOnFailure(reader.Get(span));
+        VerifyOrReturnError(span.size() < kMaxStringLen, CHIP_ERROR_INVALID_TLV_ELEMENT);
+        context.SetElementType(kElementTypeString);
+
+        std::string str(span.data(), span.size());
+        InsertKeyValue(jsonObj, context, str);
+        break;
+    }
+
+    case TLV::kTLVType_Null: {
+        context.SetElementType(kElementTypeNull);
+        InsertKeyValue(jsonObj, context, Json::Value());
+        break;
+    }
+
+    case TLV::kTLVType_Structure: {
+        Json::Value jsonStruct(Json::objectValue);
+        ReturnErrorOnFailure(TlvStructToJson(reader, jsonStruct));
+        context.SetElementType(kElementTypeStruct);
+        InsertKeyValue(jsonObj, context, jsonStruct);
+        break;
+    }
+
+    case TLV::kTLVType_Array: {
+        CHIP_ERROR err;
+        Json::Value jsonArray(Json::arrayValue);
+        const char * subType     = kElementTypeEmpty;
+        const char * nextSubType = kElementTypeEmpty;
+        size_t listIndex         = 0;
+        TLV::TLVType containerType;
+
+        ReturnErrorOnFailure(reader.EnterContainer(containerType));
+
+        while ((err = reader.Next()) == CHIP_NO_ERROR)
+        {
+            switch (reader.GetType())
+            {
+            case TLV::kTLVType_UnsignedInteger: {
+                nextSubType = kElementTypeUInt;
+                break;
+            }
+            case TLV::kTLVType_SignedInteger: {
+                nextSubType = kElementTypeInt;
+                break;
+            }
+            case TLV::kTLVType_Boolean: {
+                nextSubType = kElementTypeBool;
+                break;
+            }
+            case TLV::kTLVType_FloatingPointNumber: {
+                if (reader.IsElementDouble())
+                {
+                    nextSubType = kElementTypeDouble;
+                }
+                else
+                {
+                    nextSubType = kElementTypeFloat;
+                }
+                break;
+            }
+            case TLV::kTLVType_ByteString: {
+                nextSubType = kElementTypeBytes;
+                break;
+            }
+            case TLV::kTLVType_UTF8String: {
+                nextSubType = kElementTypeString;
+                break;
+            }
+            case TLV::kTLVType_Null: {
+                nextSubType = kElementTypeNull;
+                break;
+            }
+            case TLV::kTLVType_Structure: {
+                nextSubType = kElementTypeStruct;
+                break;
+            }
+            default:
```

Comment:
```suggestion
            case TLV::kTLVType_Array:
            default:
```

This would make it more explicit that arrays of arrays are not supported

Reply:
Except arrays _are_ supported.